APIDocumentation: Display Data Type of Parameters in API Documentation.

MSurfer20 opened the pull request:
<!-- What's this PR for?  (Just a link to an issue is fine.) -->
Fixes part of #15967 

**Testing plan:** <!-- How have you tested? -->
Tested locally on the system.
Ran tests ```./tools/test-documentation``` and ```./tools/test-api```


**GIFs or screenshots:** <!-- If a UI change.  See:
  https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
  -->
![image](https://user-images.githubusercontent.com/56172924/101791229-4eff8a80-3b29-11eb-8807-bbb89a3268a7.png)
![image](https://user-images.githubusercontent.com/56172924/101791252-532ba800-3b29-11eb-8aa4-de59ed28a450.png)


<!-- Also be sure to make clear, coherent commits:
  https://zulip.readthedocs.io/en/latest/contributing/version-control.html
  -->

timabbott commented:
@MSurfer20 thanks for working on this!  I think we could do this more attractively by just having a `boolean`, perhaps in a blue, appear in the top line just after the field name before OPTIONAL/REQUIRED/DEPRECATED.  Can you try that out and see how it looks?  

(If it's ugly, we might try moving one of these values to the upper-right corner).

timabbott commented:
Also, can you take a screenshot of how you're handling `anchor`?  That's a field with a union type, and I'm not sure how those get displayed.

MSurfer20 commented:
@timabbott Here are the screenshots of all the additional cases that are shown currently:
![image](https://user-images.githubusercontent.com/56172924/102323262-4504de00-3fa6-11eb-9ce3-99a35cdd82dd.png)
![image](https://user-images.githubusercontent.com/56172924/102323422-7a113080-3fa6-11eb-99b6-070c3116b374.png)
![image](https://user-images.githubusercontent.com/56172924/102323453-81d0d500-3fa6-11eb-805b-efcef266097a.png)
![image](https://user-images.githubusercontent.com/56172924/102323475-8ac1a680-3fa6-11eb-80e9-e40b92b73a4e.png)
![image](https://user-images.githubusercontent.com/56172924/102323643-c2305300-3fa6-11eb-8883-a03c8febe165.png)
![image](https://user-images.githubusercontent.com/56172924/102323843-03c0fe00-3fa7-11eb-9574-78d4a2756c31.png)

MSurfer20 commented:
Would we want to have a recursive loop for arrays to extract the element of the array until maximum depth, as it currently handles only ```array of X``` case, or should I handle the ```array of array``` case separately, as this is the only occurrence of such a datatype. Lastly, as currently we are displaying ```array of X``` and Union cases, having them in the top field itself might make the heading line too lengthy.

MSurfer20 commented:
@timabbott This is what the UI might look like with the changes that you suggested
![image](https://user-images.githubusercontent.com/56172924/102329139-02470400-3fae-11eb-97a8-c87dc65e4e7a.png)

andersk commented:
`array of string or integer` is unclear: it’s ambiguous between `string[] | integer` and `(string | integer)[]`, and also too easy to misread as `array or string or integer` (`unknown[] | string | integer`). I think most programmers would prefer to see the syntactic form rather than trying to parse the English form, anyway.

andersk commented:
Also, schemas are recursive; if your function to parse them isn’t recursive, it’s wrong (hence the “array of array” issue).

MSurfer20 commented:
Thanks for the review @timabbott @andersk . I have made the following changes split in two commits.
Commit 1:
1. Changed the logic to a recursive function so that cases like ```array containing (array containing (integer))``` and others are handled as well.
2. Added the CSS at appropriate place.
3. Changed ```array of string or integer``` to ```array containing (string or integer)``` for more clarity.
![image](https://user-images.githubusercontent.com/56172924/103148945-0d0e4f80-478b-11eb-9c6e-1385af5b1b87.png)
![image](https://user-images.githubusercontent.com/56172924/103148954-316a2c00-478b-11eb-89c0-cbeee7d34365.png)
![image](https://user-images.githubusercontent.com/56172924/103148999-a3db0c00-478b-11eb-9537-099b38ba015e.png)

Commit 2:
Changed the format from ```array containing (string or integer)``` to ```(string | integer)[]``` for better readability
![image](https://user-images.githubusercontent.com/56172924/103149077-48f5e480-478c-11eb-94bb-3c3b72e993dd.png)
![image](https://user-images.githubusercontent.com/56172924/103149090-5c08b480-478c-11eb-80b5-0b43bb4075b6.png)

timabbott commented:
Merged as f4cf5166bb800b5a540e145548ef06a1d2cbde9c after squashing and rewriting the commit message to focus on higher-level concerns; we rarely talk about functions added or whatnot, because that's easy to see in the code, and instead focus on the reasoning, context, etc. behind the change.

Huge thanks for doing this @MSurfer20!  This is great; a good follow-up would be to add similar output to the "Response" parameters output.

MSurfer20 commented:
Thanks @timabbott. I'll start working on the follow-up :)
